lib/src/util/dart_type_utilities.dart

```diff
@@ -560,3 +565,24 @@ class InterfaceTypeDefinition {
         library == other.library;
   }
 }
+
+extension DartTypeExtensions on DartType {
+  /// Returns the type which should be used when conducting "interface checks"
+  /// on [this].
```

Comment:
Does Dartdoc handle `[this]`? The analyzer doesn't because there isn't anywhere interesting to link it to, so we've aways written it as in-line code (I can't figure out how to escape a backquote to illustrate).

Reply:
/fyi @jcollins-g

Reply:
Reverted.